creator-node/src/contentAccess/contentAccessChecker.ts

```diff
@@ -0,0 +1,88 @@
+import { isRegisteredDiscoveryNode } from './helpers'
+import { CheckAccessArgs, CheckAccessResponse } from './types'
+import { recoverWallet, signatureHasExpired } from '../apiSigning'
+
+const CONTENT_SIGNATURE_MAX_TTL_MS = 6 * 60 * 60 * 1000 // 6 hours
+
+/**
+ * Checks that all content headers are passed in.
+ * Checks that discovery node that generated signature is registered.
+ * Checks that signature is not too old.
+ * Checks that user requesting the content has access.
+ */
+export async function checkContentAccess({
+  cid,
+  contentAccessHeaders,
+  libs,
+  logger,
+  redis
+}: CheckAccessArgs): Promise<CheckAccessResponse> {
+  if (!contentAccessHeaders) {
```

Comment:
this would no longer be headers but query params, as the DN redirect would include the content access signature in the query param.

Reply:
What will that look like exactly so I can update this

Reply:
this has been updated!